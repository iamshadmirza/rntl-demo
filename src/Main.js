import React, { useState } from 'react'
import { Stack, Card, Button, Text, Box, Header } from 'react-native-design-system';
import { ScrollView } from 'react-native'

function generateRandomString (text) {
    const array = text.split(' ');
    for (let i = array.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
    }
  return array;
}

export default function Main({ text }) {
    const [question, setQuestion] = useState(generateRandomString(text));
    const [answer, setAnswer] = useState([]);
    const [result, setResult] = useState(null)

    const selectWord = (word) => {
        const newQuestionArray = question.filter(item => item !== word);
        const newAnswerArray = [...answer, word];
        setQuestion(newQuestionArray);
        setAnswer(newAnswerArray);
        resetResult();
    }

    const unselectWord = (word) => {
        const newAnswerArray = answer.filter(item => item !== word);
        const newQuestionArray = [...question, word];
        setAnswer(newAnswerArray);
        setQuestion(newQuestionArray);
        resetResult();
    }

    const checkResult = () => {
        if (answer.join(' ') === text){
            setResult(true);
        } else {
            setResult(false);
        }
    }

    const resetResult = () => {
        setResult(null);
    }
    
    return (
        <ScrollView>
            <Header>Game Time</Header>
            <Stack style={{ margin: 10, backgroundColor: '#fefefe'}}>
                <Card shadow>
                    <Box>
                        <Text size="xlarge">Arrange the words to form a sentence</Text>
                    </Box>
                    <Card outline>
                        <Stack direction="horizontal" accessibilityLabel="questions" verticalSpace="xsmall" cropEndSpace>
                            {question.length > 0
                                ? question.map((item, index) => (
                                    <Button key={index} outline round onPress={() => selectWord(item)}>{item}</Button>
                                ))
                                : <Text>Click submit to check result</Text>
                            }
                        </Stack>
                    </Card>
                </Card>
                <Card shadow>
                    <Box>
                        <Text size="xlarge">Arranged words</Text>
                    </Box>
                    <Card outline>
                        <Stack direction="horizontal" accessibilityLabel="answers" verticalSpace="xsmall" cropEndSpace>
                            {answer.length > 0
                                ? answer.map((item, index) => (
                                    <Button key={index} outline round onPress={() => unselectWord(item)}>{item}</Button>
                                ))
                                : <Text>Select some words to show here</Text>
                            }
                        </Stack>
                    </Card>
                </Card>
                {result !== null
                    ? (
                        <Box space="xsmall">
                            <Text size="large" style={{ color: result ? 'green' : 'red'}}>
                                {result ? 'Correct answer ðŸ¥³' : 'Wrong answer ðŸ˜¢'}
                            </Text>
                        </Box> 
                    )
                    : null
                }
            </Stack>
            <Card space="xsmall">
                <Button onPress={checkResult}>Submit</Button>
            </Card>
        </ScrollView>
    );
}
