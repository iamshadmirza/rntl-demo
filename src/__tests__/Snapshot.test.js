import React from 'react';
import { create } from 'react-test-renderer';
import Main from '../Main';
import { ThemeProvider, theme } from 'react-native-design-system';

const useTheme = (children) => {
    return (
        <ThemeProvider value={theme}>
            {children}
        </ThemeProvider>
    );
}

const tree = create(useTheme(<Main text="One" />));

test('snapshot', () => {
    expect(tree).toMatchSnapshot();
});