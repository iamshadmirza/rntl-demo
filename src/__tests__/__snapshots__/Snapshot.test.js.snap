// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTScrollView>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3,
            "zIndex": 10,
          },
          Object {
            "backgroundColor": "#1e88e5",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1e88e5",
            "flexDirection": "row",
            "height": 56,
            "width": "100%",
            "zIndex": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#1e88e5",
              "color": "#fff",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "500",
              "paddingHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Game Time
        </Text>
      </View>
    </RCTSafeAreaView>
    <View
      cropEndSpace={false}
      direction="vertical"
      horizontalSpace="none"
      space="medium"
      style={
        Array [
          Object {},
          Object {
            "backgroundColor": "#fefefe",
            "margin": 10,
          },
        ]
      }
      verticalSpace="none"
    >
      <View
        style={
          Array [
            Object {
              "marginBottom": 16,
            },
            Object {
              "marginTop": 16,
            },
            Object {
              "marginHorizontal": 0,
            },
          ]
        }
      >
        <View
          outline={false}
          shadow={true}
          space="medium"
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": "#fff",
              "borderRadius": 3,
              "justifyContent": "center",
              "padding": 16,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3,
              "width": "100%",
            }
          }
        >
          <View
            background="clearWhite"
            space="medium"
            style={
              Object {
                "alignItems": "center",
                "background": "#fff",
                "justifyContent": "center",
                "padding": 16,
              }
            }
          >
            <Text
              color="default"
              fontWeight="500"
              scale={true}
              size="xlarge"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 26,
                  "fontWeight": "500",
                  "includeFontPadding": false,
                  "textAlignVertical": "center",
                }
              }
            >
              Arrange the words to form a sentence
            </Text>
          </View>
          <View
            outline={true}
            shadow={false}
            space="medium"
            style={
              Object {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderColor": "#333",
                "borderRadius": 3,
                "borderWidth": 0.5,
                "elevation": 0,
                "justifyContent": "center",
                "padding": 16,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityLabel="questions"
              cropEndSpace={true}
              direction="horizontal"
              horizontalSpace="none"
              space="medium"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexWrap": "wrap",
                  },
                  undefined,
                ]
              }
              verticalSpace="xsmall"
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 16,
                    },
                    Object {
                      "marginLeft": 16,
                    },
                    Object {
                      "marginVertical": 10,
                    },
                    Object {
                      "marginLeft": 0,
                    },
                    Object {
                      "marginRight": 0,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#1e88e510",
                        "borderColor": "#1e88e5",
                        "borderRadius": 28,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "left": 0,
                        "right": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#1e88e5",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "margin": 14,
                        }
                      }
                    >
                      One
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 16,
            },
            Object {
              "marginHorizontal": 0,
            },
          ]
        }
      >
        <View
          outline={false}
          shadow={true}
          space="medium"
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": "#fff",
              "borderRadius": 3,
              "justifyContent": "center",
              "padding": 16,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3,
              "width": "100%",
            }
          }
        >
          <View
            background="clearWhite"
            space="medium"
            style={
              Object {
                "alignItems": "center",
                "background": "#fff",
                "justifyContent": "center",
                "padding": 16,
              }
            }
          >
            <Text
              color="default"
              fontWeight="500"
              scale={true}
              size="xlarge"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 26,
                  "fontWeight": "500",
                  "includeFontPadding": false,
                  "textAlignVertical": "center",
                }
              }
            >
              Arranged words
            </Text>
          </View>
          <View
            outline={true}
            shadow={false}
            space="medium"
            style={
              Object {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderColor": "#333",
                "borderRadius": 3,
                "borderWidth": 0.5,
                "elevation": 0,
                "justifyContent": "center",
                "padding": 16,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityLabel="answers"
              cropEndSpace={true}
              direction="horizontal"
              horizontalSpace="none"
              space="medium"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexWrap": "wrap",
                  },
                  undefined,
                ]
              }
              verticalSpace="xsmall"
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 16,
                    },
                    Object {
                      "marginLeft": 16,
                    },
                    Object {
                      "marginVertical": 10,
                    },
                    Object {
                      "marginLeft": 0,
                    },
                    Object {
                      "marginRight": 0,
                    },
                  ]
                }
              >
                <Text
                  color="default"
                  fontWeight="500"
                  scale={true}
                  size="medium"
                  style={
                    Object {
                      "color": "#333",
                      "fontSize": 16,
                      "fontWeight": "500",
                      "includeFontPadding": false,
                      "textAlignVertical": "center",
                    }
                  }
                >
                  Select some words to show here
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      outline={false}
      shadow={false}
      space="xsmall"
      style={
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#fff",
          "borderRadius": 3,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#1e88e5",
              "borderColor": "#1e88e5",
              "borderRadius": 2,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "left": 0,
              "right": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#e3f2fd",
                "fontSize": 16,
                "fontWeight": "400",
                "margin": 14,
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
